### PR TITLE
Fix thread-safety issues in TrackingMonitor

### PR DESCRIPTION
**Pull request by Dr15Jones**
#### PR description:

Changing TrackingMonitor into a stream module uncovered thread-safety issues. These changes now make all activities thread-safe by avoiding direct use of the underlying ROOT objects.

#### PR validation:

The code compiles.

### CODE REVIEW

**Comment by Dr15Jones** [comment]
please test

**Comment by jfernan2** [comment]
+1

File: DQM/TrackingMonitor/src/TrackAnalyzer.cc [hunk 1]
```diff
@@ -2016,25 +2016,23 @@ void TrackAnalyzer::fillHistosForState(const edm::EventSetup& iSetup, const reco
...
+      tkmes.TkEtaPhi_Ratio_byFoldingmap->divide(tkmes.TrackEtaPhi, tkmes.TrackEtaPhiInverted, 1., 1., "");
+      tkmes.TkEtaPhi_Ratio_byFoldingmap_op->divide(tkmes.TrackEtaPhi, tkmes.TrackEtaPhiInvertedoutofphase, 1., 1., "");
 
-      int nx = tkmes.TrackEtaPhi->getTH2F()->GetNbinsX();
-      int ny = tkmes.TrackEtaPhi->getTH2F()->GetNbinsY();
+      int nx = tkmes.TrackEtaPhi->getNbinsX();
+      int ny = tkmes.TrackEtaPhi->getNbinsY();
 
+      //NOTE: for full reproducibility when using threads, this loop needs to be
+      // a critical section
```
Comment (makortel) [inline DQM/TrackingMonitor/src/TrackAnalyzer.cc]: How important is "full reproducibility when using threads"? (just thinking how soon a solution for that needs to found)

(e.g. are the divisions and get/setBinContent something that needs to be done on every event, or should they move to harvesting?)

**Comment by Dr15Jones** [comment]
@makortel I already have a good idea how to make it reproducible, even without harvesting. I didn't make that change in this pull request as I was just doing the minimum necessary to get rid of the segmentation faults.

**Comment by makortel** [comment]
Ok, great!

**Comment by silviodonato** [comment]
+1

**Comment by silviodonato** [comment]
@Dr15Jones we got many seg fault in CMSSW_11_1_X_2020-03-09-2300 (including #29143) due to  TrackingMonitor

**Comment by Dr15Jones** [comment]
@silviodonato  My code that was meant to prevent deadlocks failed so we see deadlocks in the IB. I have isolated my mistake and I have a fix coming in the next few minutes which properly handles the deadlock prevention.

**Comment by Dr15Jones** [comment]
see #29164

**Pull request merged**